### DISCUSSION TITLE
Incremental imports, repeatedly asking me to import the same album

### DISCUSSION THREAD

**Discussion by Stampede** [beetbox/beets#4843] [Q&A]
I'm importing my music library into beets. I have copy:no and move:no set in the configuration. For now, I just want to get my metadata tags correct. I don't want to actually move the files yet since that would screw up my Navidrome database.

There are 2 situations I'm running into that I am having trouble with.

1. I have a directory with a mix of songs from one artist. I'll use Dolly Parton as an example, since her material has been released and re-released so many times. Autotagger asks me what to do, I choose *t* for tracks, then it will individually ask me for each track to associate an album. I just accept the first choice and then do that for the rest of the tracks in that directory.
2. An album is a double-CD and the audio files are split between different directories. In the auto-tagger, I choose the album and note the musicbrainz release ID. Beets warns me of missing tracks and I choose "apply." Then when beets finds the other directory, I specify that same music brainz ID. Beets warns about missing tracks (that were imported when it found the first directory), I choose "apply." Beets warns me that it's a duplicate, and I tell it to merge all.

Everything seems to work OK, and the tracks show up when I do `beet ls <QUERY>` however, if I shut down the import and come back to it the next day, beets keeps making me go through the same tracks over again, even though I am doing an incremental import. I've taken to just telling it to skip when I see these over again. Am I doing something wrong with the way I import these in the first place?

I know about the --flat option, but the problem is I'm running the import over my entire music library, and the folder structure is a little inconsistent.

I tried to keep this brief; please let me know if I need to provide more information.

Thanks for any pointers!

**Comment by sampsyo** [comment]
Hmm; while I won't pretend to see the problem 100% clearly yet, it seems likely to me that incremental mode doesn't interact well with the "as Tracks" option. It may also not handle merging duplicates, although that would be more surprising to me. Is there any chance you could give it a shot on some for-testing-only directories that don't require either form of special treatment?

It may also be helpful to take a look at your configuration (the output of `beet config`).
